var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py

```diff
@@ -342,6 +343,16 @@ class IntelOneapiCompilers(IntelOneApiPackage, CompilerPackage):
     # instead.
     requires("%gcc", msg="intel-oneapi-compilers must be installed with %gcc")
 
+    # Hack for working around things like having old binutils in RHEL9.5
+    # https://access.redhat.com/solutions/7049696
+    variant(
+        "binutils",
+        default=False,
+        description="Build a binutils. (Use if unsupported opcode by OS ld/as)",
+    )
+    with when("+binutils"):
+        depends_on("binutils", type=dt.ALL_TYPES)
```

Comment:
```suggestion
    with when("+binutils"):
        depends_on("binutils", type=dt.ALL_TYPES)
```

use `with when` only if it actually covers multiple statements, and don't import private API, and correct the deptype to `type="run"`.